### PR TITLE
(Add) Model Factories and Base Tests

### PR DESCRIPTION
**Pull request by HDVinnie**
- Note this PR adds generated model factories, generated HTTP tests for controllers, generated unit tests for form requests, generated tests for console commands and configured a basic testing environment. 
- These tests are marked as incomplete!
- This PR also updates dependencies so I could use PHP 8.

```
vendor/bin/phpunit -d memory_limit=1024M
PHPUnit 8.5.21 by Sebastian Bergmann and contributors.

IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII  63 / 127 ( 49%)
IIIIIIIIII.IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII 126 / 127 ( 99%)
.                                                               127 / 127 (100%)

Time: 15.41 seconds, Memory: 58.00 MB

OK, but incomplete, skipped, or risky tests!
Tests: 127, Assertions: 6, Incomplete: 125.
```

### CODE REVIEW

**Comment by HDVinnie** [comment]
@johannac @quentincaffeino please take a look.

**Comment by HDVinnie** [comment]
Can we make this project php8 only?

**Comment by quentincaffeino** [comment]
> Can we make this project php8 only?

@HDVinnie, why do you need to upgrade it?

Besides proving that it works (these tests would do that when they are done), most of people don't use php8 yet. For me it isn't a problem since I have completely custom stack and switching version is not a problem. I know some people use cheapest hosting solutions to host this project so it would be really nice to hear from them if their hosting provider already supports php8.

**Comment by travtaz997** [comment]
> > Can we make this project php8 only?
> 
> @HDVinnie, why do you need to upgrade it?
> 
> Besides proving that it works (these tests would do that when they are done), most of people don't use php8 yet. For me it isn't a problem since I have completely custom stack and switching version is not a problem. I know some people use cheapest hosting solutions to host this project so it would be really nice to hear from them if their hosting provider already supports php8.
> 
> **Edit:** here are stats of how many actually use php8: https://blog.packagist.com/php-versions-stats-2021-1-edition/
> 
> tldr: Only around 10%.
> 
> I want this number to grow but I'm afraid such change could also hurt community.

Lurker here ... Just wanted to pop in and say I have a web hosting reseller account at HostGator and have quite a few sub-accounts.  Our only PHP options right now are 7.3 and 7.4.

image

File: database/factories/EventImageFactory.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+/* @var $factory \Illuminate\Database\Eloquent\Factory */
+
+use Faker\Generator as Faker;
+
+$factory->define(App\Models\EventImage::class, function (Faker $faker) {
+    return [
+        'image_path' => $faker->word,
```
Comment (quentincaffeino) [inline database/factories/EventImageFactory.php:9]: This could be replaced with a blank data image and ids should be replaced with real ids

File: database/factories/MyBaseModelFactory.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+/* @var $factory \Illuminate\Database\Eloquent\Factory */
+
+use Faker\Generator as Faker;
+
+$factory->define(App\Models\MyBaseModel::class, function (Faker $faker) {
```
Comment (quentincaffeino) [inline database/factories/MyBaseModelFactory.php:7]: I believe this is an abstract entity

**Comment by quentincaffeino** [comment]
All of these factories already exist:

```
AccountFactory
AccountPaymentGatewayFactory
AttendeeFactory
CountryFactory
CurrencyFactory
DateFormatFactory
DateTimeFormatFactory
DiscountCodeFactory
EventAccessCodesFactory
EventFactory
EventImageFactory
EventStatsFactory
OrderFactory
OrderItemFactory
OrderStatusFactory
OrganiserFactory
PaymentGatewayFactory
TicketFactory
TicketStatusFactory
TimezoneFactory
UserFactory
```

**Review by quentincaffeino — CHANGES_REQUESTED** [review]
Duplicate generated factories need to be removed

**Comment by HDVinnie** [comment]
> Duplicate generated factories need to be removed

No the new ones need to stay....the old ones need to be removed. 

TBH I am not sure I am going to waste time on this project. Theres only one maintainer who is clearly busy in life which is fine but there is many things in this project that need work and I don't see it happening anytime soon.

**Comment by quentincaffeino** [comment]
Old ones can't be just removed because some of them contain important code. Will need to go through them manually then and figure out what is required.

> TBH I am not sure I am going to waste time on this project. Theres only one maintainer who is clearly busy in life which is fine but there is many things in this project that need work and I don't see it happening anytime soon.

I understand, thank you for your contribution.

When you decide to drop this project I guess you can convert this pr to draft for history. If decide to use your work as a base, I also have your branch downloaded. Since I need to do tests anyways I'll get to them eventually.

**Comment by justynpride** [comment]
Wondering what the current thought on the prioritising of the creating of tests. Reviewing all of the comments on the existing pull requests, other than people time, a limitation appears to be the lack of testing built in to the current code. Should the priority of contributors not be to resolve this, so that it is easier for @johannac and others to test review and merge new features and bug fixes. I've not used it but read a lot about the benefit of Laravel Shift.
